### ISSUE TITLE
Add argument for taking to account 'Team Selected %' in squad generation and optimisation

### ISSUE THREAD

**Issue by richardthomasdev** [alan-turing-institute/AIrsenal#400]
As discussed in https://github.com/alan-turing-institute/AIrsenal/issues/384 , this issue tracks the inclusion of a new argument that will allow AIrsenal to take into account FPL player 'Team Selected %' when generating squads or running  optimisations. 

eg. `airsenal_make_squad --budget 1000 --min_teams_perc 3`

Where players fall beneath the supplied %, they will not be considered for contention in generated squads or transfer optimisations.

**Comment by nbarlowATI** [comment]
Excellent - many thanks @richardthomasdev !

I believe the "selected_by_percent" is available in the player summary data that the FPLDataFetcher can retrieve via 
```
fetcher.get_player_summary_data()
```
Note that this dict is keyed by the players "fpl_api_id" rather than their "player_id" (these two are not the same, to avoid problems when new players are added to FPL).

I would think that the new function  to check whether players pass the threshold or not could go in `airsenal/framework/optimization_utils.py` 

and then be called by the functions in `airsenal/framework/optimization_squad.py` or `airsenal/framework_optimization_pygmo.py` (for initial squad selection) and `airsenal/framework/optimization_transfers.py` (for week-to-week optimization), to decide whether or not to `add_player` to the squad.

Does that make sense?

**Comment by niall-anthony-mcnulty** [comment]
Hi guys, has this new argument been implemented already? Or is it a work in progress? Sounds great, as I did notice a lot of the 'airsenal_make_squad --num_gw 1' teams generated included players who won't be playing.

**Comment by richardthomasdev** [comment]
@niall-anthony-mcnulty this is currently work in progress :)

**Comment by jack89roberts** [comment]
We do store the number of users that have selected a player in the database (`PlayerAttributes.selected` in `airsenal.framework.schema`), but we don't do anything with it currently. Note we save the number that have selected the player though, not the percentage at the moment.

Adding something like this may fit nicely as additional optional parameters in the `list_players` function in `airsenal.framework.utils`.

**Comment by Tdarnell** [comment]
Is this currently being worked on? If not I'd happily take a look

**Comment by jack89roberts** [comment]
@Tdarnell I don't think so (unless @richardthomasdev is still working on it) - feel free to pick it up!

**Comment by richardthomasdev** [comment]
@Tdarnell I'm playing with this in my spare time but Python isn't my primary language and I'm struggling with how to cleanly implement it. I have something working that I'm not comfortable putting a pull request in for right now, so feel free! 😄

**Comment by jack89roberts** [comment]
Feel free to put a draft pull request in @richardthomasdev which you can continue to work on, and we'd be happy to help with anything you're unsure about (+ maybe @Tdarnell could contribute to your fork too). No judgement about messy or partially broken code - have plenty of that ourselves! 😄

**Comment by nbarlowATI** [comment]
This could potentially be a fun and useful thing for someone to pick up again if interested...

**Comment by Tdarnell** [comment]
I ended up being too busy to work on this but would still like to implement it. If I get the time soon I'll make a start!
